src/coreclr/jit/importer_vectorization.cpp

```diff
@@ -0,0 +1,741 @@
+        JITDUMP("impExpandHalfConstEqualsSIMD: No V256 support and data is too big for V128\n");
+        // NOTE: We might consider using four V128 for ARM64
+        return nullptr;
+    }
+
+    GenTree* zero = gtNewSimdHWIntrinsicNode(simdType, niZero, baseType, simdSize);
+
+    GenTree* offset1  = gtNewIconNode(dataOffset, TYP_I_IMPL);
+    GenTree* offset2  = gtNewIconNode(dataOffset + len * sizeof(USHORT) - simdSize, TYP_I_IMPL);
+    GenTree* dataPtr1 = gtNewOperNode(GT_ADD, TYP_BYREF, data, offset1);
+    GenTree* dataPtr2 = gtNewOperNode(GT_ADD, TYP_BYREF, gtClone(data), offset2);
+
+    GenTree* vec1 = gtNewIndir(simdType, dataPtr1);
+    GenTree* vec2 = gtNewIndir(simdType, dataPtr2);
+
+    // TODO-Unroll-CQ: Spill vec1 and vec2 for better pipelining, currently we end up emitting:
+    //
+    //   vmovdqu  xmm0, xmmword ptr [rcx+12]
+    //   vpxor    xmm0, xmm0, xmmword ptr[reloc @RWD00]
+    //   vmovdqu  xmm1, xmmword ptr [rcx+20]
+    //   vpxor    xmm1, xmm1, xmmword ptr[reloc @RWD16]
+    //
+    // While we should re-order them to be:
+    //
+    //   vmovdqu  xmm0, xmmword ptr [rcx+12]
+    //   vmovdqu  xmm1, xmmword ptr [rcx+20]
+    //   vpxor    xmm0, xmm0, xmmword ptr[reloc @RWD00]
+    //   vpxor    xmm1, xmm1, xmmword ptr[reloc @RWD16]
+    //
+
+    // ((v1 ^ cns1) | (v2 ^ cns2)) == zero
+    GenTree* xor1 = gtNewSimdBinOpNode(GT_XOR, simdType, vec1, cnsVec1, baseType, simdSize, false);
+    GenTree* xor2 = gtNewSimdBinOpNode(GT_XOR, simdType, vec2, cnsVec2, baseType, simdSize, false);
+    GenTree* orr  = gtNewSimdBinOpNode(GT_OR, simdType, xor1, xor2, baseType, simdSize, false);
+    return gtNewSimdHWIntrinsicNode(TYP_BOOL, useSingleVector ? xor1 : orr, zero, niEquals, baseType, simdSize);
+#else
+    return nullptr;
+#endif
+}
+
+//------------------------------------------------------------------------
+// impCreateCompareInd: creates the following tree:
+//
+//  *  EQ        int
+//  +--*  IND       <type>
+//  |  \--*  ADD       byref
+//  |     +--*  <obj>
+//  |     \--*  CNS_INT   <offset>
+//  \--*  CNS_INT   <value>
+//
+// Arguments:
+//    comp -       Compiler object
+//    obj -        GenTree representing data pointer
+//    type -       type for the IND node
+//    offset -     offset for the data pointer
+//    value -      constant value to compare against
+//
+// Return Value:
+//    A tree with indirect load and comparison
+//
+static GenTree* impCreateCompareInd(Compiler* comp, GenTree* obj, var_types type, ssize_t offset, ssize_t value)
+{
+    GenTree* offsetTree    = comp->gtNewIconNode(offset, TYP_I_IMPL);
+    GenTree* addOffsetTree = comp->gtNewOperNode(GT_ADD, TYP_BYREF, obj, offsetTree);
+    GenTree* indirTree     = comp->gtNewIndir(type, addOffsetTree);
+    GenTree* valueTree     = comp->gtNewIconNode(value, type);
+    if (varTypeIsSmall(indirTree))
+    {
+        indirTree = comp->gtNewCastNode(TYP_INT, indirTree, true, TYP_UINT);
+        valueTree->ChangeType(TYP_INT);
+    }
```

Comment:
This cast is a nop.

Also, `ChangeType` on a freshly created constant tree...?